pkg/controller/controlplane/valuesprovider.go

```diff
@@ -493,7 +493,18 @@ func (vp *valuesProvider) GetControlPlaneChartValues(
 		return nil, err
 	}
 
-	authValues, err := getAuthNGroupRoleChartValues(cpConfig, cluster, vp.controllerConfig.Auth)
+	metalEndpoint := metalControlPlane.Endpoint
+	metalCredentials, err := metalclient.ReadCredentialsFromSecretRef(ctx, vp.client, &cp.Spec.SecretRef)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You'd save one lookup if you this before the metal client initialization and then use `NewClientFromCredentials` for initializing the metal client.

Reply:
That's a good point, done